src/librustc_mir/borrow_check/nll/constraints/graph.rs

```diff
@@ -196,3 +210,56 @@ impl<'s, 'graph, D: ConstraintGraphDirecton> graph::GraphSuccessors<'graph> for
     type Item = RegionVid;
     type Iter = Successors<'graph, D>;
 }
+
+crate struct AugmentedRegionGraph<'s> {
+    graph: NormalRegionGraph<'s>,
+    static_region: RegionVid,
+}
+
+impl<'s> AugmentedRegionGraph<'s> {
+    crate fn new(
+        graph: NormalRegionGraph<'s>,
+        static_region: RegionVid,
+    ) -> Self {
+        AugmentedRegionGraph {
+            graph,
+            static_region,
+        }
+    }
+}
+
+impl<'s> graph::DirectedGraph for AugmentedRegionGraph<'s> {
+    type Node = RegionVid;
+}
+
+impl<'s> graph::WithNumNodes for AugmentedRegionGraph<'s> {
+    fn num_nodes(&self) -> usize {
+        self.graph.constraint_graph.first_constraints.len()
+    }
+}
+
+impl<'s> graph::WithSuccessors for AugmentedRegionGraph<'s> {
+    fn successors<'graph>(
+        &'graph self,
+        node: Self::Node,
+    ) -> <Self as graph::GraphSuccessors<'graph>>::Iter {
+        let static_successors = if node == self.static_region {
+            Some(self.static_region..self.graph.constraint_graph.first_constraints.len().into())
+        } else {
+            None
+        };
+
+        let normal_successors = self.graph.outgoing_regions(node).collect::<Vec<_>>();
+
+        Box::new(
+            static_successors
+            .into_iter()
+            .flat_map(|range| range)
+            .chain(normal_successors))
+    }
+}
+
+impl<'s, 'graph> graph::GraphSuccessors<'graph> for AugmentedRegionGraph<'s> {
+    type Item = RegionVid;
+    type Iter = Box<dyn Iterator<Item = Self::Item>>;
```

Comment:
I couldn't figure out how to name this type so I just boxed it. Should I hand roll a custom `Iterator` instead?